### PR TITLE
Fix bug sf360. Plot associations button does nothing

### PR DESCRIPTION
**Pull request by gbm19**
Fix of sf #360 bug  for the Qt4 version

### CODE REVIEW

**Comment by alxpl** [comment]
Confirmed, I just applied the patch in different versions of Fedora and the bug is fixed.

File: libscidavis/src/CurvesDialog.cpp [hunk 1]
```diff
@@ -212,10 +212,15 @@ void CurvesDialog::showPlotAssociations()
 		curve = 0;
 
     ApplicationWindow *app = (ApplicationWindow *)this->parent();
```
Comment (highperformancecoder) [inline libscidavis/src/CurvesDialog.cpp:214]: Can we be guaranteed that the parent (if there is one) is an ApplicationWindow? Better to use a dynamic_cast here, I would think...

Comment (gbm19) [reply libscidavis/src/CurvesDialog.cpp:214]: In code there is only one use of the CurvesDialog constructor: In ApplicationWindow.cpp at line 5258:
`CurvesDialog* crvDialog = new CurvesDialog(this);`

Comment (highperformancecoder) [reply libscidavis/src/CurvesDialog.cpp:214]: That may be the case now, but there's nothing enforcing that assumption. For instance, someone may write a unittest for CurvesDialog...

I'm, merging the PR, but will change that line to a dynamic_cast. There's no performance reason for not using a dynamic_cast here.

**Pull request merged**